url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/cmdlib/LoadLibrary.java

```diff
@@ -0,0 +1,251 @@
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.sdicons.scripty.cmdlib;
+
+import com.sdicons.scripty.parser.CommandException;
+import com.sdicons.scripty.parser.IContext;
+import com.sdicons.scripty.parser.IEval;
+import com.sdicons.scripty.spec.type.ITypeSpec;
+import com.sdicons.scripty.spec.type.TypeSpecException;
+import com.sdicons.scripty.annot.*;
+import com.sdicons.scripty.repl.ReplEngine;
+import com.sdicons.scripty.repl.ReplEngineException;
+import com.sdicons.scripty.spec.type.TypeUtil;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * <ul>
+ *    <li><b><code>load</code></b> Load and execute one or more scripts. If the file exists and is readable, it will be rememberd for the reload.<code>(load file | classpath:/resource | cp:/resource ...)</code></li>
+ *    <li><b><code>reload</code></b> Reload previously loaded files. A file is remembered if it existed and was readable. <code>(reload)</code></li>
+ * </ul>
+ */
+public class LoadLibrary
+{
+    private List<Loader> loaders = new ArrayList<Loader>();
+
+    @ScriptyCommand(name="load")
+    @ScriptyVarArgList(vararg = @ScriptyArg(name="loaders", type="Custom com.sdicons.scripty.cmdlib.LoadLibrary$LoaderType"))
+    public void load(@ScriptyParam("loaders") Object[] aLoaders, IContext aCtx, IEval aEval)
+    throws CommandException
+    {
+        List<Loader> lLoaders = new ArrayList<Loader>();
+        for(Object lLoader : aLoaders)
+        {
+            lLoaders.add((Loader)lLoader);
+        }
+        internalLoad(lLoaders, aEval, aCtx);
+    }
+    
+    @ScriptyCommand(name="reload")
+    @ScriptyStdArgList()
+    public void reload(IContext aContext, IEval aEval)
+    throws CommandException
+    {
+        List<Loader> lLoaders = new ArrayList<Loader>();
+        lLoaders.addAll(loaders);
+        internalLoad(lLoaders, aEval, aContext);
+    }
+    
+    private void internalLoad(List<Loader> aLoaders, IEval aEval, IContext aContext)
+    throws CommandException
+    {
+        Object lInput = aContext.getBinding(ReplEngine.INPUT);
+        Object lOutput = aContext.getBinding(ReplEngine.OUTPUT);
+        Object lError = aContext.getBinding(ReplEngine.ERROR);
+   
+        final ReplEngine lEngine = new ReplEngine();
+        lEngine.setContext(aContext);
+        lEngine.setCommandRepository(aEval.getCommandRepo());
+        lEngine.setMacroRepository(aEval.getMacroRepo());
+        
+        try
+        {
+            for(Loader lLoader: aLoaders)
+            {
+                loaders.remove(lLoader);
+
+                InputStream lIn = null;
+                try
+                {
+                    lIn = lLoader.getStream();
+                    lEngine.startNonInteractive(lIn);
+                }
+                catch (ReplEngineException e)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 106.